lib/widgets/settings.dart

```diff
@@ -54,3 +56,26 @@ class _ThemeSetting extends StatelessWidget {
       ]);
   }
 }
+
+class _BrowserPreferenceSetting extends StatelessWidget {
+  const _BrowserPreferenceSetting();
+
+  void _handleChange(BuildContext context, bool newOpenLinksWithInAppBrowser) {
+    GlobalStoreWidget.of(context).updateGlobalSettings(
+      GlobalSettingsCompanion(browserPreference: Value(
+        newOpenLinksWithInAppBrowser ? BrowserPreference.inApp
+                                     : BrowserPreference.external)));
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final zulipLocalizations = ZulipLocalizations.of(context);
+    final openLinksWithInAppBrowser =
+      GlobalStoreWidget.of(context).globalSettings.urlLaunchMode
+      == UrlLaunchMode.platformDefault;
```

Comment:
Given that we'll be presenting the setting as either "in-app: yes" or "in-app: no", let's make the semantics of our default match that presentation. IOW on Android let's adjust our default behavior so that it's the same as the behavior we end up with if the database says `inApp`.

That'd be good to do as a prep commit, before introducing the setting.